src/PrestaShopBundle/Form/Admin/Sell/Product/FooterType.php

```diff
@@ -50,35 +51,46 @@ class FooterType extends TranslatorAwareType
      */
     private $router;
 
+    /**
+     * @var ProductPreviewProvider
+     */
+    protected $productPreviewUrlProvider;
+
     /**
      * @param TranslatorInterface $translator
      * @param array $locales
      * @param ProductProvider $productUrlProvider
+     * @param ProductPreviewProvider $productPreviewUrlProvider
      * @param RouterInterface $router
      */
     public function __construct(
         TranslatorInterface $translator,
         array $locales,
         ProductProvider $productUrlProvider,
+        ProductPreviewProvider $productPreviewUrlProvider,
         RouterInterface $router
     ) {
         parent::__construct($translator, $locales);
         $this->productUrlProvider = $productUrlProvider;
         $this->router = $router;
+        $this->productPreviewUrlProvider = $productPreviewUrlProvider;
     }
 
     /**
      * {@inheritDoc}
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $request = Request::createFromGlobals();
+        $page = $request->query->get('page');
```

Comment:
Each time I see such code my hair stands up 😅. Using the request is usually a bad practice and very error-prone in my experience And heaven more when it's not injected but created with static methods this is a last resort solution that should be avoided as much as possible

And in this case I don't understand why it's needed to get back a parameter from the URL, because the link "Cancel" can simply be hard-coded with the edit route `admin_products_v2_edit`